cdap-ui/app/cdap/components/PipelineConfigurations/ConfigurationsContent/ConfigurationsActionButtons/index.js

```diff
@@ -81,48 +100,99 @@ ConfigActionButtons.propTypes = {
   validToSave: PropTypes.bool,
   pipelineEdited: PropTypes.bool,
   updatingPipeline: PropTypes.bool,
-  saveAndClose: PropTypes.func
+  updatingPipelineAndAction: PropTypes.bool,
+  saveAndClose: PropTypes.func,
+  saveAndAction: PropTypes.func,
+  actionLabel: PropTypes.string
 };
 
 const ConnectedConfigActionButtons = connect(mapStateToProps)(ConfigActionButtons);
 
 export default class ConfigurationsActionButtons extends Component {
   state = {
-    updatingPipeline: false
+    // need 2 states here instead of just 1, to determine which button to show
+    // spinning wheel on
+    updatingPipeline: false,
+    updatingPipelineAndAction: false
   };
 
   static propTypes = {
     onClose: PropTypes.func,
-    activeTab: PropTypes.string
+    activeTab: PropTypes.string,
+    action: PropTypes.func
+  };
+
+  static defaultProps = {
+    action: runPipeline
+  };
+
+  close = () => {
+    if (typeof this.props.onClose === 'function') {
+      this.props.onClose();
+    }
+  };
+
+  closeAndAction = () => {
```

Comment:
The name seems too abstract. Just by looking at the name I don't know what the action this function is performing. May be if there are multiple actions just suffix it with specific action so that it becomes easier to understand?